### PR TITLE
#166604318 Fix current category not pre-populated with current category before update

### PR DESCRIPTION
**Pull request by hnobi**
#### What Does This PR Do?
This PR pre-populate the update event form with the current event category

#### Description Of Task To Be Completed
- add react life cycle to update the state with props

#### How can this be manually tested?
- `git pull mvp-develop`
- `git checkout bg-fix-pre-populate-category-166604318` and `make start`
- click on any event created by you and click on the update button,
 you will see the category field pre-populated

#### What are the relevant pivotal tracker stories?
[#166604318](https://www.pivotaltracker.com/story/show/166604318)

#### Screenshot(s)
Nil

### CODE REVIEW

**Review by Alameen688 — COMMENTED** [review]
Nice work @hnobi. However, this implementation doesn't update the category dropdown selected value on change (i.e even though the actual value is changed the rendered value doesn't change).

Kindly help look into that. Thanks.

File: client/components/common/CustomDropDown.js [hunk 1]
```diff
@@ -14,11 +14,19 @@ class CustomDropDown extends Component {
     super(props);
     this.state = {
       listOpen: false,
-      headerTitle: this.props.title,
+      headerTitle: '',
```
Comment (Alameen688) [inline client/components/common/CustomDropDown.js]:
```suggestion
      headerTitle: this.props.title
```
This should be left as is, due to the fact that the dropdown still needs an initial text. Which is supplied to it by the parent component.

For more context, the "filters" section on the events page which also uses this component has no initial text is rendered due to this.

Kindly help look into that, thanks

Comment (hnobi) [reply client/components/common/CustomDropDown.js]: okay

**Review by Alameen688 — APPROVED** [review]
👍 Works as expected!

**Pull request merged**